### DISCUSSION TITLE
how to calculate the gene activity matrix for integrated scATAC-seq

### DISCUSSION THREAD

**Discussion by xiaowenchenjax** [stuart-lab/signac#575] [Q&A] [answered]
Hi Tim,
After integrating all 6 scATAC-seq libraries, I would like to calculate the gene activity matrix for integrated scATAC-seq, however, this is seurat object, I met the following error 
**Annotation(integrated) <- annotations
Error in UseMethod(generic = "Annotation<-", object = object) :
  no applicable method for 'Annotation<-' applied to an object of class "Assay"**

What I should do? Thanks.

**Comment by timoast** [comment, marked as answer]
The integrated assay here is a standard Seurat assay, rather than a ChromatinAssay. You will need to change the default assay to the original ATAC assay (not the integrated assay), which should be a ChromatinAsasy.

**Reply by xiaowenchenjax** [reply]
Thank you. Tim
this is my integrated object.
> integrated
An object of class Seurat
1514939 features across 31012 samples within 5 assays
Active assay: new_assay (217588 features, 0 variable features)
 4 other assays present: peaks, RNA, peaks1, integrated
 2 dimensional reductions calculated: integratedLSI, umap

peaks1 assay is the integrated one, peaks assay is the original ATAC assay. under peaks assay, I do have gene activity matrix (i.e. RNA). The integration step also update this  matrix? I thought i should use counts matrix under peaks1 assay to create a new ChromatinAssay to re-calculate gene activity matrix.

**Reply by timoast** [reply]
>  under peaks assay, I do have gene activity matrix (i.e. RNA). The integration step also update this matrix?

I don't quite understand what you mean that you have a gene activity matrix in the peaks assay. Did you create an assay containing the gene activities and name it "peaks"?

> The integration step also update this matrix?

See the documentation for `Seurat::IntegrateData()`

> I thought i should use counts matrix under peaks1 assay to create a new ChromatinAssay to re-calculate gene activity matrix.

I don't know what all the different assays you have are and how they were created, so it'd difficult to advise what to do here. Basically, you can either compute a gene activity matrix for each object before integrating the data, or you can change the active assay back to the original, un-corrected, ChromatinAssay and run `GeneActivity()` on the integrated object.
